envbuilder.go

```diff
@@ -77,152 +76,9 @@ var (
-	FallbackImage string `env:"FALLBACK_IMAGE"`
-
-	// ExitOnBuildFailure terminates the container upon a build failure.
-	// This is handy when preferring the `FALLBACK_IMAGE` in cases where
-	// no devcontainer.json or image is provided. However, it ensures
-	// that the container stops if the build process encounters an error.
-	ExitOnBuildFailure bool `env:"EXIT_ON_BUILD_FAILURE"`
-
-	// ForceSafe ignores any filesystem safety checks.
-	// This could cause serious harm to your system!
-	// This is used in cases where bypass is needed
-	// to unblock customers!
-	ForceSafe bool `env:"FORCE_SAFE"`
-
-	// Insecure bypasses TLS verification when cloning
-	// and pulling from container registries.
-	Insecure bool `env:"INSECURE"`
-
-	// IgnorePaths is a comma separated list of paths
-	// to ignore when building the workspace.
-	IgnorePaths []string `env:"IGNORE_PATHS"`
-
-	// SkipRebuild skips building if the MagicFile exists.
-	// This is used to skip building when a container is
-	// restarting. e.g. docker stop -> docker start
-	// This value can always be set to true - even if the
-	// container is being started for the first time.
-	SkipRebuild bool `env:"SKIP_REBUILD"`
-
-	// GitURL is the URL of the Git repository to clone.
-	// This is optional!
-	GitURL string `env:"GIT_URL"`
-
-	// GitCloneDepth is the depth to use when cloning
-	// the Git repository.
-	GitCloneDepth int `env:"GIT_CLONE_DEPTH"`
-
-	// GitCloneSingleBranch clones only a single branch
-	// of the Git repository.
-	GitCloneSingleBranch bool `env:"GIT_CLONE_SINGLE_BRANCH"`
-
-	// GitUsername is the username to use for Git authentication.
-	// This is optional!
-	GitUsername string `env:"GIT_USERNAME"`
-
-	// GitPassword is the password to use for Git authentication.
-	// This is optional!
-	GitPassword string `env:"GIT_PASSWORD"`
-
-	// GitHTTPProxyURL is the url for the http proxy.
-	// This is optional!
-	GitHTTPProxyURL string `env:"GIT_HTTP_PROXY_URL"`
-
-	// WorkspaceFolder is the path to the workspace folder
-	// that will be built. This is optional!
-	WorkspaceFolder string `env:"WORKSPACE_FOLDER"`
-
-	// SSLCertBase64 is the content of an SSL cert file.
-	// This is useful for self-signed certificates.
-	SSLCertBase64 string `env:"SSL_CERT_BASE64"`
-
-	// ExportEnvFile is an optional file path to a .env file where
-	// envbuilder will dump environment variables from devcontainer.json and
-	// the built container image.
-	ExportEnvFile string `env:"EXPORT_ENV_FILE"`
-
-	// PostStartScriptPath is the path to a script that will be created by
-	// envbuilder based on the `postStartCommand` in devcontainer.json, if any
-	// is specified (otherwise the script is not created). If this is set, the
-	// specified InitCommand should check for the presence of this script and
-	// execute it after successful startup.
-	PostStartScriptPath string `env:"POST_START_SCRIPT_PATH"`
-
+type Dependencies struct {
```

Comment:
Why do we need a separate struct here?

Reply:
yes, but as separate function arguments, not a dedicated `Deps` wrapper.